### PR TITLE
EZP-27259: Repository and components skeleton

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-27259

# Description

This PR initializes the repository with the skeletons for `ez-platform-ui-app`, `ez-navigation-hub` and `ez-toolbar`. At the moment, there is more meta-information and meta-files than actual code ;-)

Still a few words on some choices:

* this repository mostly follows Polymer (good?) practices in terms of structure (ie the element definitions are in the root of the repository, the test in `test/`, we also have `demo/` directory). The only exception to that is the *externalization* of JavaScript files in dedicated `.js` files instead of being integrated into `.html`. In any case, this is something that can easily be fixed if this happens to be a mistake.
* the unit tests are located into `test/` and should have the same structure as the code being tested.
* there's no *task manager* (gulp, grunt, ...), for now `npm` scripts seems to be enough:
  * `npm test` to run the unit test
  * `npm run lint` to run code style checks
* on code style, it's a good opportunity to switch to eslint which is the reference in that area. I tried to create a configuration more or less compatible with the code we were writing in PlatformUIBundle (except that it forces you to use ES6 `let` or `const` instead of `var`). But for new code, we could also switch to a *standard* coding style instead of having our own. @yannickroger @bdunogier @andrerom @StephaneDiot any opinion on that ?

## Tasks

* [x] Skeletons for `ez-platform-ui-app`, `ez-navigation-hub` and `ez-toolbar`
* [x] Unit tests skeletons
* [x] Demo skeletons
* [ ] ~~API Doc~~ (waiting for *Polymer 2.0 way* of doing it...)
* [x] Travis configuration
* [x] eslint integration

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+dist: trusty
+language: node_js
+node_js: stable
```
Comment (dpobel) [inline .travis.yml]: self note: consider using .nvmrc instead

File: README.md [hunk 2]
```diff
@@ -1 +1,9 @@
 # hybrid-platform-ui-core-components
+
+[![Build Status](https://travis-ci.org/ezsystems/hybrid-platform-ui-core-components.svg?branch=master)](https://travis-ci.org/ezsystems/hybrid-platform-ui-core-components)
+
+Provides the following custom elements used in the Hybrid Platform UI:
+
+* \<ez-platform-ui-app\>
+* \<ez-navigation-hub\>
+* \<ez-toolbar\>
```
Comment (dpobel) [inline README.md]: self note: add system requirements and how to run unit test and linting

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,315 @@
+{
```
Comment (bdunogier) [inline .eslintrc.json:1]: You had fun writing this, had you not ? :)

File: README.md [hunk 4]
```diff
@@ -1 +1,9 @@
 # hybrid-platform-ui-core-components
+
+[![Build Status](https://travis-ci.org/ezsystems/hybrid-platform-ui-core-components.svg?branch=master)](https://travis-ci.org/ezsystems/hybrid-platform-ui-core-components)
+
+Provides the following custom elements used in the Hybrid Platform UI:
+
+* \<ez-platform-ui-app\>
```
Comment (bdunogier) [inline README.md]: monospace font ?

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "ezsystems/hybrid-platform-ui-core-components",
+    "description": "Hybrid Platform UI core frontend components",
+    "type": "ezplatform-assets",
+    "license": "GPL-v2",
```
Comment (bdunogier) [inline composer.json]: The syntax is `GPL-2.0` on our other repositories, for what it's worth.

**Review by bdunogier — CHANGES_REQUESTED** [review]
Minor details, besides that, +1.

File: .eslintrc.json (hunk 3 above)
Comment (dpobel) [reply .eslintrc.json:1]: lol it was more or less auto-generated from PlatformUIBundle code base actually :)

**Comment by andrerom** [comment]
>  any opinion on that ?

I agree, and if eslint is now the norm it makes sense to rather use that then jslint/jshint.

**Comment by dpobel** [comment]
For now, I think we'll keep on using a standard close to what we were writing PlatformUIBundle with one exception due to the use of ES6 `const`. So instead of writing something like:

```js
var foo,
    bar,
    CONSTANT_FOO = 'foo',
    CONSTANT_BAR = 'bar';
```

we'll now use:

```js
let foo;
let bar;
const CONSTANT_FOO = 'foo';
const CONSTANT_BAR = 'bar';
```

so basically, several `const` / `let` should be used.
Reasoning:
* this still guarantee to generate a minimal diff in case of change
* this avoids weird indentation when defining several constant with `const`

(gentle reminder ping @StephaneDiot @yannickroger)

**Comment by yannickroger** [comment]
Quick comment to say I had a (late) look at it and it is great.

**Pull request merged**